orttraining/orttraining/training_ops/cpu/op_gradients.cc

```diff
@@ -260,5 +260,39 @@ Status QuickGeluGrad<T>::Compute(OpKernelContext* context) const {
   return Status::OK();
 }
 
+ONNX_OPERATOR_KERNEL_EX(LeakyReluGrad, kMSDomain, 1, kCpuExecutionProvider,
+                        KernelDefBuilder().TypeConstraint("T", DataTypeImpl::GetTensorType<float>()),
+                        LeakyReluGrad<float>);
+
+template <typename T>
+Status LeakyReluGrad<T>::Compute(OpKernelContext* context) const {
+  auto& dY = *context->Input<Tensor>(0);
+  const T* dY_data = dY.template Data<T>();
+  auto& Y = *context->Input<Tensor>(1);
+  const T* Y_data = Y.template Data<T>();
+  auto& dX = *context->Output(0, dY.Shape());
+  T* dX_data = dX.template MutableData<T>();
+
+  concurrency::ThreadPool* tp = context->GetOperatorThreadPool();
+  int64_t elem_count = dY.Shape().Size();
+  constexpr int64_t length_per_task = 4096;  // this number comes from FastGelu.
+  int64_t task_count = (elem_count + length_per_task - 1) / length_per_task;
+
+  concurrency::ThreadPool::TryBatchParallelFor(
+      tp, static_cast<int32_t>(task_count),
+      [&](ptrdiff_t task_idx) {
+        const auto start = task_idx * length_per_task;
+        const T* p_dy = dY_data + start;
+        const T* p_y = Y_data + start;
+        T* p_dx = dX_data + start;
+        int64_t count = std::min(length_per_task, elem_count - start);
+        for (int64_t i = 0; i < count; i++) {
+          p_dx[i] = p_dy[i] * (p_y[i] > 0.0f ? 1.0f : alpha_);
```

Comment:
Could not find a way to use `Eigen` `Map` with conditional

`dx = dy * (y > 0 ? 1.0f : alpha)`

Let me know if you are aware of any way to do that.

Reply:
Not sure if the sample code in onnxruntime/core/providers/cpu/activation/activation.h can help or not, something like, which use ".select()" function:

```
  void operator()(std::ptrdiff_t first, std::ptrdiff_t last) const final {
    ptrdiff_t len = last - first;
    T* output_ptr = this->output + first;
    ConstEigenVectorArrayMap<T> xm(this->input + first, len);
    EigenVectorArrayMap<T> ym(output_ptr, len);
    ym = (xm >= 0).select(xm, (T)alpha * (xm.exp() - 1));
  }
```

Reply:
Thanks. :)